app/models/x509_validator.rb

```diff
@@ -46,8 +46,20 @@ def certificate_key_is_supported(record, x509)
   end
 
   def certificate_is_strong(record, x509)
-    return if x509.public_key.params['n'].num_bits >= 2048
+    if x509.public_key.params['n'].num_bits >= 2048
+      certificate_digest_right(record, x509)
+    else
+      record.errors.add(:certificate, I18n.t('certificates.errors.small_key'))
+    end
+  end
+
+  def valid_algorithms
+    %w[sha256WithRSAEncryption sha384WithRSAEncryption sha512WithRSAEncryption]
+  end
+
+  def certificate_digest_right(record, x509)
```

Comment:
```suggestion
  def hash_algorithm_is_strong(record, x509)
```
maybe? to be consistent...